apps/prairielearn/src/ee/lib/ai-grading/ai-grading-test.ts

```diff
@@ -0,0 +1,435 @@
+import assert from 'node:assert';
+
+import * as cheerio from 'cheerio';
+import { OpenAI } from 'openai';
+import { zodResponseFormat } from 'openai/helpers/zod';
+import { z } from 'zod';
+
+import * as error from '@prairielearn/error';
+import {
+  loadSqlEquiv,
+  queryAsync,
+  queryOptionalRow,
+  queryRow,
+  queryRows,
+  runInTransactionAsync,
+} from '@prairielearn/postgres';
+
+import { config } from '../../../lib/config.js';
+import {
+  type AssessmentQuestion,
+  type Course,
+  IdSchema,
+  InstanceQuestionSchema,
+  type Question,
+  RubricItemSchema,
+  SubmissionGradingContextEmbeddingSchema,
+} from '../../../lib/db-types.js';
+import * as manualGrading from '../../../lib/manualGrading.js';
+import { buildQuestionUrls } from '../../../lib/question-render.js';
+import { getQuestionCourse } from '../../../lib/question-variant.js';
+import { createServerJob } from '../../../lib/server-jobs.js';
+import * as questionServers from '../../../question-servers/index.js';
+
+import * as aiGradingUtil from './ai-grading-util.js';
+
+const sql = loadSqlEquiv(import.meta.url);
+
+export async function aiGradeTest({
+  course,
+  course_instance_id,
+  question,
+  assessment_question,
+  urlPrefix,
+  authn_user_id,
+  user_id,
+}: {
+  question: Question;
+  course: Course;
+  course_instance_id?: string;
+  assessment_question: AssessmentQuestion;
+  urlPrefix: string;
+  authn_user_id: string;
+  user_id: string;
+}): Promise<string> {
+  // If OpenAI API Key and Organization are not provided, throw error
+  if (!config.openAiApiKey || !config.openAiOrganization) {
+    throw new error.HttpStatusError(403, 'Not implemented (feature not available)');
+  }
+  const openai = new OpenAI({
+    apiKey: config.openAiApiKey,
+    organization: config.openAiOrganization,
+  });
+
+  const question_course = await getQuestionCourse(question, course);
+
+  const serverJob = await createServerJob({
+    courseId: course.id,
+    courseInstanceId: course_instance_id,
+    assessmentId: assessment_question.assessment_id,
+    authnUserId: authn_user_id,
+    userId: user_id,
+    type: 'ai_grading_test',
+    description: 'Test accuracy for AI grading',
+  });
+
+  serverJob.executeInBackground(async (job) => {
```

Comment:
I'm just sort of squinting at things, but this all seems almost identical to what's happening in `aiGrade()`. Is there any possibility of code reuse here? I'm normally fine with small amounts of duplication but this looks like hundreds of lines of almost the same code.